src/content_script/mastodonInject.js

```diff
@@ -129,20 +129,26 @@ async function injectInteractionButtons() {
 async function init() {
     injectFollowButton();
 
-    const observer = new MutationObserver(() => {
-        Promise.allSettled([
-            injectInteractionButtons(),
-        ]);
-    });
+    const ogType = document.querySelector("meta[property='og:type']");
 
-    const feedElement = await waitForElement(
-        "#mastodon .item-list[role='feed']",
-        false,
-        TIMEOUT_DURATION
-    );
-    observer.observe(feedElement, {
-        childList: true, subtree: true,
-    });
+    if (ogType && ogType.getAttribute("content") === "article"){
+        injectInteractionButtons();
+    } else {
+        const observer = new MutationObserver(() => {
+            Promise.allSettled([
+                injectInteractionButtons(),
+            ]);
+        });
+
+        const feedElement = await waitForElement(
+            "#mastodon .item-list[role='feed']",
+            false,
+            TIMEOUT_DURATION
+        );
+        observer.observe(feedElement, {
```

Comment:
BTW I noticed a nullable problem here: If `waitForElement` runs into the timeout, then what does the `MutationObserver` do? It get's an `observer.observe(null, {`, so listens on a null element?

That seems useless and problematic at least. Whether or not this may be related to our problem, it certainly is no proper error handling. I'd suggested to e.g. throw an `Error` and catch/log it or so…

Reply:
AH okay, or not, we want to stay silent on non Mastodon sites… though maybe would be useful for debugging here hmm, I am unsure.